### PR TITLE
refactor: yeet job queues scaffolding in favor of only graphile worker

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Problem

Since the first iteration of jobs was built, we've kept in a ton of overhead code (job queue classes, job queue manager) on the off-chance that we'd eventually implement a tiered job queues mechanism (e.g. memory, Postgres, S3). 

We've decided against that architecture now, so let's yeet all the unnecessary code.

## Changes

This is a pure refactor that aims to get rid of all the job queue scaffolding in favor of a simple `graphileWorker`. This is also going to make it much easier/cleaner to soon implement the graphile cron.

Couldn't realistically keep this small, so I've tried to keep the commits very concise in case it makes it easier to review.

## How did you test this code?

Wrapping up for today, but tomorrow I'll write tests for `GraphileWorker`. 

I've manually tested this works for now.

### CODE REVIEW

File: plugin-server/src/main/jobs/local/mock-graphile-worker.ts [hunk 1]
```diff
@@ -16,17 +14,25 @@ interface FsJob {
     pluginConfigId?: number
     pluginConfigTeam?: number
 }
-export class FsQueue extends JobQueueBase {
-    paused: boolean
-    started: boolean
+export class MockGraphileWorker {
```
Comment (yakkomajuri) [inline plugin-server/src/main/jobs/local/mock-graphile-worker.ts:17]: Just ignore this for now - it's a whole other refactor that needs to be done and I'm trying to keep this PR concise.

I want to do the minimal amount of work to make this "mock generic job queue work" but will likely later do away with this as it's not needed. Used in quite a few tests for now

**Comment by yakkomajuri** [comment]
TODO:

- [x] Bring back async start and stop functionality
- [x] Write tests for graphileWorker

**Comment by yakkomajuri** [comment]
There may or may not be a couple tests failing but marking this ready for review as it's effectively done! 

cc @hazzadous

File: plugin-server/src/main/ingestion-queues/anonymous-event-buffer-consumer.ts [hunk 2]
```diff
@@ -79,7 +79,10 @@ export const startAnonymousEventBufferConsumer = async ({
 
             status.debug('⬆️', 'Enqueuing anonymous event for processing', { job })
             try {
-                await graphileQueue.enqueue(JobName.BUFFER_JOB, job)
+                await graphileWorker.enqueue(JobName.BUFFER_JOB, job, {
```
Comment (hazzadous) [inline plugin-server/src/main/ingestion-queues/anonymous-event-buffer-consumer.ts:82]: I think the answer is yes as retryOnFailure is false, but this call will fail on PG failure right?

**Review by hazzadous — APPROVED** [review]
Happy with this. The main thing is that I _think_ the `enqueue` will raise on pg connection errors.

File: plugin-server/src/main/ingestion-queues/anonymous-event-buffer-consumer.ts (hunk 2 above)
Comment (yakkomajuri) [reply plugin-server/src/main/ingestion-queues/anonymous-event-buffer-consumer.ts:82]: the behavior remains the same as before in this case - i.e. if retries are enabled we don't throw, else we do. That means we will throw in the anonymous event consumer

**Comment by yakkomajuri** [comment]
Confirmed with @tiina303 that the test failing is a flake. She's working on this test suite in #12126

File: plugin-server/tests/main/jobs/graphile-worker.test.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            expect(runRetriableFunctionArgs.metricTags).toEqual({ jobName: 'pluginJob' })
+            expect(runRetriableFunctionArgs.tryFn).not.toBeUndefined()
+            expect(runRetriableFunctionArgs.catchFn).not.toBeUndefined()
+            expect(runRetriableFunctionArgs.finallyFn).toBeUndefined()
+        })
+    })
+
+    describe('syncState()', () => {
+        it('creates a new runner if necessary', async () => {
+            jest.spyOn(graphileWorker, 'createPool').mockImplementation(() =>
```
Comment (macobo) [inline plugin-server/tests/main/jobs/graphile-worker.test.ts:60]: Note: mockResolveValue exists!

**Pull request merged**